vac/raw/mix.md

```diff
@@ -889,9 +910,442 @@ are fixed strings and MUST be agreed upon across implementations.
+This fixed-length encoding ensures that packets remain indistinguishable on
+the wire and prevents correlation attacks based on routing metadata structure.
+
+Implementations MAY use any address and delay encoding format agreed upon
+by all participating mix nodes, as long as the combined length is exactly
+$tκ$ bytes. The encoding format MUST be interpreted consistently by all
+nodes within a deployment.
+
+For interoperability, a recommended default encoding format involves:
+
+- Encoding the next-hop or destination libp2p multi-address:
+  - Each mix node is limited to one IPv4 circuit relay multiaddress to keep the
+  next-hop routing block compact while allowing relay connectivity. This
+  ensures that most nodes can act as mix nodes, including those behind NATs
+  or firewalls.
+  - In libp2p terms, this combines transport addresses with multiple peer
+  identities to form an address that describes a relay circuit:  
+  `
+  /ip4/<ipv4>/tcp/<port>/p2p/<relayPeerID>/p2p-circuit/p2p/<relayedPeerID>
+  `  
+  Variants may include directly reachable peers and transport such as
+  `/quic-v1`, depending on the mix node's supported stack.
+  - IPv6 support is deferred, as it adds $16$ bytes just for the IP field.
+  - Future revisions may extend this format to support IPv6 or DNS-based
+  multiaddresses.
+  
+  With these constraints, the recommended encoding layout is:
+  - IPv4 address (4 bytes)
+  - Protocol identifier _e.g.,_ TCP or QUIC (1 byte)
+  - Port number (2 bytes)
+  - Peer IDs (39 bytes, post-Base58 decoding)
+
+- Encoding the forwarding delay as an unsigned 16-bit integer (2 bytes) in
+  milliseconds, using big endian network byte order.
+
+If the encoded address or delay is shorter than its respective allocated
+field, it MUST be padded with zeros. If it exceeds the allocated size, it
+MUST be rejected or truncated according to the implementation policy.
+
+With the field sizes and encoding conventions established, the next section describes
+how a mix node constructs a complete Sphinx packet when initiating the Mix Protocol.
+
+### 8.5 Packet Construction
+
+This section defines how a mix node constructs a Sphinx packet when initiating
+the Mix Protocol on behalf of a local origin protocol instance.
+The construction process wraps the message in a sequence of encryption
+layers&mdash;one for each hop&mdash;such that only the corresponding mix node
+can decrypt its layer and retrieve the routing instructions for that hop.
+
+#### 8.5.1 Inputs
+
+To initiate the Mix Protocol, the origin protocol instance submits a message
+to the Mix Entry Layer on the same node. This layer forwards it to the local
+Mix Protocol instance, which constructs a Sphinx packet
+using the following REQUIRED inputs:
+
+- **Application message**: The serialized message provided by the origin
+  protocol instance. The Mix Protocol instance applies any configured spam
+  protection mechanism and attaches one or two SURBs prior to encapsulating
```

Comment:
Just to check my understanding: multiple SURBs are just to provide return path redundancy?

Reply:
Do we want to include a placeholder link here that will link to a future SURB section?